asttokens/asttokens.py

```diff
@@ -285,87 +349,58 @@ def get_text_positions(self, node, padded, unmarked=False):
 
     return start, end
 
-  def _get_text_positions_unmarked(self, node, padded):
-    # type: (AstNode, bool) -> Tuple[Tuple[int, int], Tuple[int, int]]
-    """
-    Unmarked version of ``get_text_positions()``.
 
-    Raises an error for unsupported Python versions, unlike ``init_tokens=False``.
-    Raises an error for astroid tree.
-    Doesn't raise an error for unsupported types of AST node, but may return incorrect results.
-    """
-    # supports_unmarked() already checks the Python version, but writing it this way
-    # also tells mypy about the version. This prevents errors below with end_lineno and end_col_offset.
-    if sys.version_info[:2] < (3, 8) or not supports_unmarked():
-      raise NotImplementedError('Python version not supported')
-
-    if not isinstance(node, (ast.AST, type(None))):
-      raise NotImplementedError('Not supported for astroid')
-
-    if isinstance(node, ast.Module):
-      # Modules don't have position info, so just return the range of the whole text.
-      # The token-using method does something different, but its behavior seems weird and inconsistent.
-      # For example, in a file with only comments, it only returns the first line.
-      # It's hard to imagine a case when this matters.
-      return (1, 0), self._line_numbers.offset_to_line(len(self._text))
-
-    if not hasattr(node, 'lineno'):
-      return (1, 0), (1, 0)
+class LazyHelper(TextRangeBase, object):
+  def __init__(self, source_text, tree=None, filename='<unknown>'):
+    # type: (Any, Optional[Module], str) -> None
+    # FIXME: Strictly, the type of source_type is one of the six string types, but hard to specify with mypy given
+    # https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
 
-    assert node  # tell mypy that node is not None, which we allowed up to here for compatibility
+    if not isinstance(tree, (ast.AST, type(None))):
+      raise NotImplementedError('LazyHelper only supports AST trees')
```

Comment:
I suspect with tweaks to how the `unmarked` parameters are handled this limitation could be dropped. Having it keeps the tests essentially unchanged and it'd be easier to remove the limitation later than to add it back in.